gips/0081-indexing-payments.md

```diff
@@ -0,0 +1,165 @@
+
+The proposed payment system minimizes trust in both directions.
+
+When combined with the cryptoeconomic security from stake and appropriate fault-detection, the protocol can establish a cost-of-corruption, or a cost to faults in general. With this in place, a new participant entering the Network, for example as a Gateway or even as a Consumer running their own Gateway, can use the Network to its full potential. Any trusted relationship enshrined at the protocol level would directly jeopardize the Network's censorship-resistance and resilience, as it could require participants to interact with only a handful of trusted participants.
+
+These principles are evident in the original protocol design and in the ongoing efforts to roll out the Timeline Aggregation Protocol (see GIP-0054). The design of the payments mechanism in this GIP is designed to preserve these principles, while allowing maximum flexibility for Gateways to provide alternative payment systems for data consumers or subgraph developers.
+
+Moreover, the payment mechanism for this GIP is designed in a way that allows extension to arbitrary data services. The initial implementation is subgraphs-specific, but the Escrow and voucher system can allow arbitrary subscription-like payments, for any verifiable use case where an initial agreement leads to periodic payments of up to a certain amount for work that can be represented by an onchain proof or commitment.
+
+It works as follows:
+
+- **Gateways** will add funds to an **Escrow** account specific to the Indexer they want to interact with. This prevents double spending (the Gateway could otherwise pay itself from those same funds before the Indexer collects payment). Funds are locked with a thawing period, so the Indexer can know that funds will be available for at least that amount of time.
+- When the Gateway wants an Indexer to perform indexing work, it sends a signed **Voucher**. The Voucher specifies the max cost per period, the agreed price per gas/unit of work, and the details of the work to do (i.e. which subgraph).
+- The Indexer can post that Voucher onchain, which results in accepting the **Agreement**, the terms for which are encoded in the Voucher. The Voucher specifies that the Subgraph Service contract is authorized to release the payment; the Subgraph Service contract also specifies the stake-to-fees ratio. From the moment the voucher is posted onchain, the Indexer is guaranteed to be able to collect payment if they post the PoI (indexing up to the current Epoch) within the specified time and meeting all necessary conditions (e.g. provisioned stake).
+- The Gateway doesn’t need to sign anything else to release further payments, they just need to keep the Escrow topped up. The Indexer can stop working if they don’t see sufficient funds in the Escrow. This ensures that Gateways cannot avoid payment after the Indexer has done the work.
```

Comment:
Reframing the last sentance, as original form was not true in general, only when relying on Escrow.
```suggestion
- The Gateway doesn’t need to sign anything else to release further payments, they just need to keep the Escrow topped up. The Indexer can stop working if they don’t see sufficient funds in the Escrow. Indexers know that escrowed funds are availble to collect payment for work completed, without relying on trust.
```

Reply:
153 bracketed comment is different to double spending?

Reply:
>The Indexer can post that Voucher onchain, which results in accepting the **Agreement**,

Could be more explicit about the indexer needing to post the voucher onchain in order to accept the agreement